rcl_logging_spdlog/src/rcl_logging_spdlog.cpp

```diff
@@ -133,8 +133,8 @@ rcl_logging_ret_t rcl_logging_external_initialize(
     char * logdir = nullptr;
     rcl_logging_ret_t dir_ret = rcl_logging_get_logging_directory(allocator, &logdir);
     if (RCL_LOGGING_RET_OK != dir_ret) {
-      // We couldn't get the log directory, so get out of here without setting up
-      // logging.
+      // We couldn't get the log directory, so exit without setting up logging.
+      rcutils_reset_error();
       RCUTILS_SET_ERROR_MSG("Failed to get logging directory");
```

Comment:
can we do the following instead of resetting the message? that is gonna give more information for the caller when they collect the error message?

```suggestion
      // We couldn't get the log directory, so exit without setting up logging.
      RCUTILS_SET_ERROR_MSG_AND_APPEND_PREV_ERROR("Failed to get logging directory");
```

Reply:
Good call, applied in 0a234c2